src/Secret.java

```diff
@@ -1,21 +1,77 @@
-import java.util.Random;
-
 public class Secret {
-	private char[] code;
-	
-	public Secret() {
-		Random codeGenerator = new Random();
-		int number = codeGenerator.nextInt(10000);
-		String newCode = String.valueOf(number); 
-		this.setCode(newCode.toCharArray());
+
+	private String digits;
+	private final int LENGTH_OF_DIGITS = 4;
+
+	public Secret(String digits) {
+		// Either pad or remove digits to enforce length of 4
+		this.digits = createDigits(digits);
```

Comment:
I like that you've started with this, it should remain here until we can justify that creation of the Secret in this way is needed and who should be responsible. 

I think we know that padding substrings is not an inherent quality of a 'real' Secret in our domain, but is sort of necessary machinery for making our software work in practice.

Let's stick with this until we work on the Output/Printer boundary together, and then the input/Secret boundary may yield new ideas.